frontend/src/pages/detail/PODetail.tsx

```diff
@@ -0,0 +1,251 @@
+import React, { FormEvent } from "react";
+import { Button } from "primereact/button";
+import { InputText } from "primereact/inputtext";
+import { ToggleButton } from "primereact/togglebutton";
+import { Calendar, CalendarProps } from "primereact/calendar";
+import { Dropdown, DropdownProps } from "primereact/dropdown";
+import { DataTable } from "primereact/datatable";
+import { TableColumn } from "../../components/Table";
+import { Column, ColumnEditorOptions } from "primereact/column";
+import { InputNumber } from "primereact/inputnumber";
+import ConfirmButton from "../../components/ConfirmButton";
+
+interface modifyPOState {
+  date: any;
+  vendor: string;
+  checked: boolean;
+  confirmationPopup: boolean;
+}
+
+interface Vendors {
+  name: string;
+  code: string;
+}
+
+const dataVendors: Vendors[] = [
+  { name: "New York", code: "NY" },
+  { name: "Rome", code: "RM" },
+  { name: "London", code: "LDN" },
+  { name: "Istanbul", code: "IST" },
+  { name: "Paris", code: "PRS" },
+];
+
+interface POlineRow {
+  books: string;
+  quantity: number;
+  unitRetailPrice: number;
+}
+
+const dataPO: POlineRow[] = [
+  {
+    books: "blah",
+    quantity: 20,
+    unitRetailPrice: 3.9,
+  },
+  {
+    books: "ohaha",
+    quantity: 200,
+    unitRetailPrice: 4.0,
+  },
+];
+
+const columns: TableColumn[] = [
+  { field: "books", header: "Books", filterPlaceholder: "Books" },
+  { field: "quantity", header: "Quantity", filterPlaceholder: "Quantity" },
+  {
+    field: "unitRetailPrice",
+    header: "Unit Retail Price",
+    filterPlaceholder: "Price",
+  },
+];
+
+class ModifyPOPage extends React.Component<{}, modifyPOState> {
+  constructor(props = {}) {
+    super(props);
+    this.state = {
+      date: new Date(),
+      vendor: "asdfa",
+      checked: false,
+      confirmationPopup: false,
+    };
+  }
+
+  isPositiveInteger = (val: any) => {
+    let str = String(val);
+
+    str = str.trim();
+
+    if (!str) {
+      return false;
+    }
+
+    str = str.replace(/^0+/, "") || "0";
+    const n = Number(str);
+
+    return n !== Infinity && String(n) === str && n >= 0;
+  };
+
+  onCellEditComplete = (e: {
+    rowData: any;
+    newValue: any;
+    field: any;
+    originalEvent: any;
+  }) => {
+    const { rowData, newValue, field, originalEvent: event } = e;
+
+    switch (field) {
```

Comment:
Same comments here as on the BookAdd just define your own interface to replace TableColumn, and use the cellEditValidator for each column rather than the if statements